pkg/test/deployment/helm.go

```diff
@@ -19,59 +19,73 @@ import (
-	return map[string]string{
-		"global.tag":                                 s.Images.Tag,
-		"global.hub":                                 s.Images.Hub,
-		"global.imagePullPolicy":                     string(s.Images.ImagePullPolicy),
-		"global.proxy.enableCoreDump":                boolString(s.EnableCoreDump),
-		"global.mtls.enabled":                        boolString(s.GlobalMtlsEnabled),
-		"galley.enabled":                             boolString(s.GalleyEnabled),
-		"gateways.istio-ingressgateway.replicaCount": uintString(s.IngressGateway.ReplicaCount),
-		"gateways.istio-ingressgateway.autoscaleMin": uintString(s.IngressGateway.AutoscaleMin),
-		"gateways.istio-ingressgateway.autoscaleMax": uintString(s.IngressGateway.AutoscaleMax),
-	}
+const (
+	namespaceTemplate = `apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+  labels:
+    istio-injection: disabled
+`
+)
+
+// HelmConfig configuration for a Helm-based deployment.
+type HelmConfig struct {
+	Accessor   *kube.Accessor
+	KubeConfig string
+	Namespace  string
+	WorkDir    string
+	ChartDir   string
+
+	// Can be either a file name under ChartDir or an absolute file path.
+	ValuesFile string
+	Values     map[string]string
 }
 
-func newHelmDeployment(s *Settings, a *kube.Accessor, chartDir string, valuesFile valuesFile) (*Instance, error) {
+// NewHelmDeployment creates a new Helm-based deployment instance.
+func NewHelmDeployment(c HelmConfig) (*Instance, error) {
 	instance := &Instance{}
 
-	instance.kubeConfig = s.KubeConfig
-	instance.namespace = s.Namespace
+	instance.kubeConfig = c.KubeConfig
+	instance.namespace = c.Namespace
 
 	// Define a deployment name for Helm.
-	deploymentName := fmt.Sprintf("%s-%v", s.Namespace, time.Now().UnixNano())
+	deploymentName := fmt.Sprintf("%s-%v", c.Namespace, time.Now().UnixNano())
 	scopes.CI.Infof("Generated Helm Instance name: %s", deploymentName)
 
-	instance.yamlFilePath = path.Join(s.WorkDir, deploymentName+".yaml")
+	instance.yamlFilePath = path.Join(c.WorkDir, deploymentName+".yaml")
 
-	vFile := path.Join(chartDir, string(valuesFile))
+	// Convert the valuesFile to an absolute file path.
+	valuesFile := c.ValuesFile
+	if _, err := os.Stat(valuesFile); os.IsNotExist(err) {
+		valuesFile = filepath.Join(c.ChartDir, valuesFile)
+		if _, err := os.Stat(valuesFile); os.IsNotExist(err) {
+			return nil, err
+		}
+	}
 
 	var err error
 	var generatedYaml string
 	if generatedYaml, err = HelmTemplate(
 		deploymentName,
-		s.Namespace,
-		env.IstioChartDir,
-		vFile,
-		s); err != nil {
+		c.Namespace,
+		c.ChartDir,
+		valuesFile,
+		c.Values); err != nil {
 		return nil, fmt.Errorf("chart generation failed: %v", err)
 	}
 
 	// TODO: This is Istio deployment specific. We may need to remove/reconcile this as a parameter
 	// when we support Helm deployment of non-Istio artifacts.
-	namespaceData := fmt.Sprintf(namespaceTemplate, s.Namespace)
+	namespaceData := fmt.Sprintf(namespaceTemplate, c.Namespace)
```

Comment:
This means we will be creating the namespace for each Helm based deployment implicitly, right?

Reply:
Yeah that's right.  Nothing has changed WRT that.